Azure/communication-ui-library-android
# [Feature] feature flags made private

## Purpose

* Introduce "internal" library that is brought in by implementation (gradle will hide from public API)
* Migrate FeatureFlags to this library

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test the code

```
Remove internal project from demo-app build.gradle dependencies, verify that FeatureFlag errors show up (Internal not passed through)
```

## What to Check
Verify that the following are valid
* ...

## Other Information
